### PR TITLE
Cert user rework merge

### PR DESCRIPTION
**Pull request by Jatewo**
## Description

A rework of the User model as seen in the MODELS.md in the resources repo. The user will now have two "subprofiles" pointing to it in the database - a student profile for the app, and a content-creator profile for web.

## Changes

- Deleted credentialsRoutes and ContentCreatorRoutes (+relating tests), as these are now redundant
- Added error codes (see resources repo)
- Moved ensureNewValues to validation.js
- Renamed ContentCreatorApplication to ContentCreator
- Removed basic user info from contentCreator model
- Created Student model for app users and moved all Student specific information to this model from User
- Made a reference to a baseuser for contentCreator and Student models
- Made small updates to authRoutes
- Made StudentRoutes and moved all Student specific routes to this file (these are accessed at '/api/students'
- Changed UserRoute to create basic profiles for Students and ContentCreators when registering as a user
- Updated /api/courses/creator/:id to find a creator via a baseUser id
- Updated ref in Courses
- Removed unused imports
- Updated fakeContentCreator to take baseUserId
- Added checks for id's validity and updated tests to check for this
- Removed some console.logs
- Updated courseRoutes tests to use the new user model

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Notes for reviewers

Please check that your work works as intended. I have tried to change as little as possible but changes were required in quite a few places.

### CODE REVIEW

File: __tests__/fixtures/fakeUser.js [hunk 1]
```diff
@@ -11,6 +11,5 @@ module.exports = function makeFakeUser(email = 'fake@gmail.com', attempts = [])
         firstName: 'Fake first name',
         lastName: 'Fake last name',
         resetAttempts: attempts,
-        subscriptions: []
```
Comment (Mafusn) [inline __tests__/fixtures/fakeUser.js:14]: Why is this removed?

File: models/ContentCreator.js
Comment (Mafusn) [inline models/ContentCreator.js:1]: Name of file should be plural

**Review by Mafusn — CHANGES_REQUESTED** [review]
Small comments, looks good overall.

File: __tests__/fixtures/fakeUser.js (hunk 1 above)
Comment (Jatewo) [reply __tests__/fixtures/fakeUser.js:14]: is no longer included in the User model

File: models/ContentCreator.js
Comment (Jatewo) [reply models/ContentCreator.js:1]: I don't think we have any guidelines for this but it is probably the easiest considering the API guidelines

File: models/Students.js
[the remark this answers is not in the document]
Comment (Jatewo) [reply models/Students.js:1]: Is this currently in Dev? I don't think I have changed anything (at least not on purpose). And to clarify: have we removed isComplete from courses and sections?

**Review by Mafusn — APPROVED** [review]

**Comment by JackSMunn** [comment]
The base is changed to staging for whatever reason, so we need to change it back to dev

**Pull request merged**